runtime/caching/cache.go

```diff
@@ -19,21 +20,27 @@ type Constructor func(ctx Context, options interface{}) (Resource, error)
 // the option of being an exclusive or shared-cache, meaning if resources may be
 // re-used before they are released.
 type Cache struct {
-	m           sync.Mutex
-	shared      bool
-	entries     []*cacheEntry
-	constructor Constructor
-	tracker     gc.ResourceTracker
+	m               sync.Mutex
+	shared          bool
+	entries         []*cacheEntry
+	constructor     Constructor
+	tracker         gc.ResourceTracker
+	cachehit        runtime.Monitor
+	constructortime runtime.Monitor
+	countCachehits  float64
 }
 
 // New returns a Cache wrapping constructor such that resources
 // returned from Require are shared between all calls to Require with the same
 // options, if shared is set to true. Otherwise, resources are exclusive.
-func New(constructor Constructor, shared bool, tracker gc.ResourceTracker) *Cache {
+func New(constructor Constructor, shared bool, tracker gc.ResourceTracker, monitor runtime.Monitor) *Cache {
 	return &Cache{
-		constructor: constructor,
-		tracker:     tracker,
-		shared:      shared,
+		constructor:     constructor,
+		tracker:         tracker,
+		shared:          shared,
+		cachehit:        monitor.WithPrefix("cache-hit"),
```

Comment:
Store the monitor and do `WithPrefix` when reporting... in fact you won't need it as `.Count(` takes a name